frontend/javascripts/admin/voxelytics/task_list_view.tsx

```diff
@@ -0,0 +1,598 @@
+import React, { useEffect, useState, useMemo } from "react";
+import {
+  Collapse,
+  Input,
+  Row,
+  Col,
+  Menu,
+  Button,
+  Dropdown,
+  notification,
+  message,
+  Tag,
+  Tooltip,
+} from "antd";
+import {
+  ClockCircleOutlined,
+  MinusCircleOutlined,
+  CloseCircleOutlined,
+  SyncOutlined,
+  CheckCircleOutlined,
+  ExclamationCircleOutlined,
+  LeftOutlined,
+} from "@ant-design/icons";
+import MiniSearch from "minisearch";
+import ColorHash from "color-hash";
+
+import { Link, useLocation, useParams } from "react-router-dom";
+import moment from "moment";
+import { useUpdateEvery } from "libs/react_hooks";
+import {
+  VoxelyticsRunState,
+  VoxelyticsTaskConfig,
+  VoxelyticsTaskConfigWithHierarchy,
+  VoxelyticsTaskConfigWithName,
+  VoxelyticsTaskInfo,
+  VoxelyticsWorkflowReport,
+} from "types/api_flow_types";
+import { getVoxelyticsLogs } from "admin/admin_rest_api";
+import { formatDateMedium, formatDistance, formatDistanceStrict } from "libs/format_utils";
+import DAGView from "./dag_view";
+import TaskView from "./task_view";
+import { formatLog } from "./log_tab";
+
+const { Panel } = Collapse;
+const { Search } = Input;
+
+function getFilteredTasks(
+  miniSearch: MiniSearch<any>,
+  report: VoxelyticsWorkflowReport,
+  searchString: string,
+): Array<VoxelyticsTaskConfigWithName> {
+  const allTasks: Array<VoxelyticsTaskConfigWithName> = Object.keys(report.config.tasks).map(
+    (key) => ({
+      ...report.config.tasks[key],
+      taskName: key,
+    }),
+  );
+
+  if (searchString.length > 0) {
+    const results = miniSearch.search(searchString, {
+      fuzzy: 0.2,
+      prefix: true,
+      combineWith: "AND",
+    });
+    const resultTaskNames = results.map((result) => result.id);
+    return allTasks.filter((task) => resultTaskNames.includes(task.taskName));
+  }
+
+  return allTasks;
+}
+
+/**
+ * Adapted from https://stackoverflow.com/questions/486896/adding-a-parameter-to-the-url-with-javascript
+ * Add a URL parameter (or changing it if it already exists)
+ */
+function addUrlParam(location: ReturnType<typeof useLocation>, key: string, val: string) {
+  const search = new URLSearchParams(location.search);
+  search.append(key, val);
+  return `${location.pathname}?${search.toString()}`;
+}
+
+function removeUrlParam(location: ReturnType<typeof useLocation>, key: string) {
+  const search = new URLSearchParams(location.search);
+  search.delete(key);
+  return `${location.pathname}?${search.toString()}`;
+}
+
+function TaskStateTag({ taskInfo }: { taskInfo: VoxelyticsTaskInfo }) {
+  useUpdateEvery(10000);
```

Comment:
```suggestion
  // Re-render every 10s so that the durations that are relative to the current time
  // are updated regularly.
  useUpdateEvery(10000);
```

took me a moment to figure out why this is necessary.